hack/cherry-pick-pull.sh

```diff
@@ -200,6 +208,10 @@ for pull in "${PULLS[@]}"; do
 
   # remove the patch file from /tmp
   rm -f "/tmp/${pull}.patch"
+
+  # get the release notes
+  notes=$(curl "https://api.github.com/repos/${MAIN_REPO_ORG}/${MAIN_REPO_NAME}/pulls/${pull}" -sS | jq '.body' | grep -Po "\`\`\` *${RELEASE_NOTE_CATEGORY} ${RELEASE_NOTE_TARGET_GROUP}.*?\`\`\`")
```

Comment:
On macOs this requires gnu grep.

You can probably improve install-requirements with

```diff
$ git diff
diff --git a/hack/install-requirements.sh b/hack/install-requirements.sh
index 5b5044bff..e673322f3 100755
--- a/hack/install-requirements.sh
+++ b/hack/install-requirements.sh
@@ -45,11 +45,12 @@ Please make sure you have installed the following requirements:
 - GNU Parallel

 Brew command:
-$ brew install coreutils gnu-tar gnu-sed jq parallel
+$ brew install coreutils gnu-tar gnu-sed jq parallel grep

 Please allow them to be used without their "g" prefix:
 $ export PATH=/usr/local/opt/coreutils/libexec/gnubin:\$PATH
 $ export PATH=/usr/local/opt/gnu-tar/libexec/gnubin:\$PATH
 $ export PATH=/usr/local/opt/gnu-sed/libexec/gnubin:\$PATH
+$ export PATH=/usr/local/opt/grep/libexec/gnubin:\$PATH
 EOM
 fi
```

Reply:
Thanks!
I have added it (also sorted the packages alphabetically).